src/shim.ts

```diff
@@ -25,7 +25,7 @@ export function orchestrator(fn: (context: IOrchestrationFunctionContext) => Ite
 export function entity(fn: (context: IEntityFunctionContext) => unknown)
     : (context: IEntityFunctionContext) => void {
     const listener = new Entity(fn).listen();
-    return (context: IEntityFunctionContext) => {
-        listener(context);
+    return async (context: IEntityFunctionContext) => {
```

Comment:
I'm a little confused here, because it looks like we are returning a type of `(context: IEntityFunctionContext) => Promise<void>) instead of `(context: IEntityFunctionContext) => void).  I'm not sure why typescript doesn't yell at us here, and what the implications are of us returning that new type instead.

Assuming the end to end works, I would recommend we just update the return type of `entity` to our new return type.

Reply:
As far as I've been able to determine the returned type didn't really make a different, but I'm not 100% sure as to how the trigger is invoked and whether the trigger is concerned about the type.